src/views/order_list_view.js

```diff
@@ -0,0 +1,108 @@
+import Backbone from 'backbone';
+import OrderView from './order_view';
+import Order from '../models/order';
+
+const OrderListView = Backbone.View.extend({
+  initialize(params) {
+    this.template = params.template;
+    this.bus = params.bus;
+    this.quotes = params.quoteList;
+
+    this.listenTo(this.model, 'update', this.render);
+    this.listenTo(this.bus, 'quote_change', this.buyOrSellOrder);
+
+    this.displayOrderForm();
+  },
+
+  render() {
+    this.$('#orders').empty();
+
+    this.model.forEach((order) => {
+        const orderView = new OrderView({
+          model: order,
+          template: this.template,
+          tagName: 'li',
+          className: 'order',
+          bus: this.bus,
+        });
+      this.$('#orders').append(orderView.render().$el);
+    });
+
+    return this;
+  },
+
+  events: {
+    'click button.btn-buy': 'buyOrder',
+    'click button.btn-sell': 'sellOrder',
+  },
+
+  displayOrderForm() {
+    this.quotes.each((quote) => {
+      const symbol = quote.get('symbol');
+      this.$('.order-entry-form select').append(`<option value="${symbol}">${symbol}</option>`);
+    });
+  },
+
+  getFormData(action) {
+    const orderData = {};
+
+    orderData['symbol'] = this.$('.order-entry-form select[name="symbol"] option:selected').val();
+    orderData['quote'] = this.quotes.findWhere({symbol: orderData['symbol']});
+    orderData['targetPrice'] = Number(this.$('.order-entry-form input[name="price-target"]').val());
+    orderData['buy'] = action;
+
+    return orderData;
+  },
+
+  clearFormData() {
+    this.$('.order-entry-form input[name="price-target"]').val('');
+  },
+
+  buyOrder(event) {
+    this.addOrder(event, 'buy');
+  },
+
+  sellOrder(event) {
+    this.addOrder(event, 'sell');
+  },
+
+  addOrder(event, action) {
+    event.preventDefault();
+
+    const formData = this.getFormData(action);
+
+    const newOrder = new Order(formData);
+    this.model.add(newOrder);
+    this.clearFormData();
+  },
+
+  buyOrSellOrder(quote) {
+    let currentOrders = this.model.where({symbol: quote.get('symbol')});
+    currentOrders.forEach((order) => {
+
+      if (order.get('buy') == 'buy' && order.get('targetPrice') >= quote.get('price'))  {
+        let tradeItem = {
+          price: order.get('targetPrice'),
+          symbol: order.get('symbol'),
+          buy: 'bought',
+        }
+        this.bus.trigger('boughtOrSold', tradeItem);
+        order.destroy();
+        quote.buy();
+      }
+
+      else if (order.get('buy') == 'sell' && order.get('targetPrice') <= quote.get('price')) {
+        let tradeItem = {
+          price: order.get('targetPrice'),
+          symbol: order.get('symbol'),
+          buy: 'sold',
+        }
+        this.bus.trigger('boughtOrSold', tradeItem)
+        order.destroy();
+        quote.sell();
```

Comment:
Currently, the workflow for executing an open order looks like this:

1. Quote price changes
1. `quote_price_change` event on the bus
1. Handled here by `OrderListView.buySellQuote()`
1. Find `Order`s matching the quote
1. For each one, determine if it needs to be sold

But we can simplify this! The key observation is that when you create the `Order` you must know which `Quote` it's for - you need this information to validate the `Order`. Instead of working through the intermediaries of the bus and the `OrderListView`, each `Order` could listen directly to its `Quote` for price changes. Then the workflow would look like:

1. Quote price changes
1. Event on the quote
1. Handler in `Order`
1. Determine if the order needs to be sold

This eliminates dependencies on the bus and the `OrderListView`. It also simplifies the code, since we don't need to work with a list of `Order`s. If there are multiple `Orders` corresponding to the same quote each will have a separate handler registered, and each will run when the event occurs (steps 3-4).